km/km_coredump.c

```diff
@@ -968,7 +968,8 @@ int km_dump_core(char* core_path,
       sprintf(cpath, "%s.conf", core_path);
       if ((cfd = open(cpath, O_RDWR | O_CREAT | O_TRUNC, 0600)) < 0) {
          km_warn("Cannot create %s '%s'", "snapshot config", cpath);
-         return errno;
+         (void)unlink(cpath);
```

Comment:
I meant for the unlink() to be done down at the out: label. Sorry I wasn't clear.
unlinking here seems pointless because the open failed.